tests/php/Core/ObjectTest.php

```diff
@@ -329,6 +329,39 @@ public function testRemoveExtension()
         );
     }
 
+    public function testRemoveExtensionPreservesParentExtensions()
+    {
+        ObjectTest\ExtensionRemoveTestSubclass::add_extension(ExtendTest2::class);
+        $this->assertTrue(
+            ObjectTest\ExtensionRemoveTestSubclass::has_extension(ExtendTest1::class),
+            'Extensions should be inherited from parent classes'
+        );
+        $this->assertTrue(
+            ObjectTest\ExtensionRemoveTestSubclass::has_extension(ExtendTest2::class),
+            'Extensions added to a subclass should be registered to the subclass'
+        );
+
+        ObjectTest\ExtensionRemoveTestSubclass::remove_extension(ExtendTest2::class);
+        $this->assertTrue(
+            ObjectTest\ExtensionRemoveTestSubclass::has_extension(ExtendTest1::class),
+            'Non-specified extensions inherited from a parent class should not be removed from subclass'
+        );
+        $this->assertFalse(
+            ObjectTest\ExtensionRemoveTestSubclass::has_extension(ExtendTest2::class),
+            'Specified extensions should be removed from specified class'
+        );
+
+        ObjectTest\ExtensionRemoveTestSubclass::remove_extension(ExtendTest1::class);
+        $this->assertFalse(
+            ObjectTest\ExtensionRemoveTestSubclass::has_extension(ExtendTest1::class),
+            'Specified extensions inherited from parent class should be removed from specified class'
```

Comment:
As per my comment in the discussion, I'm not sure I agree with this chain. It should be an error IMO to try to remove a parent class's extension from a subclass.

Reply:
Yep this is fair enough, I can remove this from the tests